clang/lib/Basic/Targets/PPC.cpp

```diff
@@ -828,10 +829,7 @@ const TargetInfo::AddlRegName GCCAddlRegNames[] = {
 };
 
 ArrayRef<TargetInfo::AddlRegName> PPCTargetInfo::getGCCAddlRegNames() const {
-  if (ABI == "elfv2")
-    return llvm::ArrayRef(GCCAddlRegNames);
-  else
-    return TargetInfo::getGCCAddlRegNames();
+  return llvm::ArrayRef(GCCAddlRegNames);
```

Comment:
This blanket addition of the vs registers for all ABIs makes me a little nervous. I'm not sure that all ABIs will (or do) support these registers. For example, we support ELFv1 but based on this: https://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi.html#REG
I'm not sure it supports the vs registers.

Since we are doing this for AIX it may be safer just to explicitly mention XCOFF.

Reply:
@stefanp-ibm, GCC does accept these register names for ELFv1: https://github.com/llvm/llvm-project/pull/68476#discussion_r1349529109

For the compiler, I think the main responsibility would be to understand that these registers may overlay non-volatile FP and VMX registers.

Reply:
Accepting an non-exist(future defined) register in the asm clobber list should not cause any issue in the compiler side, I think. The register allocator can not allocate these future defined registers anyhow.

The more serious issue is that clang accepts vsx instructions in asm block even for targets that does not have VSX, see https://godbolt.org/z/rYnMbPPTE (gcc has same behavior.)

Reply:
Okay fair enough. If GCC accepts there on ELFv1 then we can too.